src/js/app.js

```diff
@@ -70,102 +70,101 @@ function getPokemon() { //(latitude, longitude) {
-					    console.log("removed pokemon at index " + i)
-					    delete allNearbyPokemon[i];
-					    //allNearbyPokemon.splice(i, 1);
-					  }
+				allNearbyPokemon = allNearbyPokemon.filter( function( el ){ return (typeof el !== "undefined"); } );
+
+				// Assemble dictionary using our keys
+				var dictionary = {};
+
+				// take closest 9 (or fewer if not available; sentinel indicated by pokemonId == 0)
+				var j;
+				for (j = 0; j < 9; j++) {
+
+					if (j < allNearbyPokemon.length) {
+						dictionary["Pokemon" + (j + 1) + "Id"] = allNearbyPokemon[j].pokemonId;
+						dictionary["Pokemon" + (j + 1) + "ExpirationTime"] = allNearbyPokemon[j].pokemonExpirationTime;
+						dictionary["Pokemon" + (j + 1) + "Distance"] = allNearbyPokemon[j].pokemonDistance;
+						dictionary["Pokemon" + (j + 1) + "Bearing"] = allNearbyPokemon[j].pokemonBearing;
+					} else {
+						dictionary["Pokemon" + (j + 1) + "Id"] = 0;
+						dictionary["Pokemon" + (j + 1) + "ExpirationTime"] = 0;
+						dictionary["Pokemon" + (j + 1) + "Distance"] = 0;
+						dictionary["Pokemon" + (j + 1) + "Bearing"] = 0;
+						break;
 					}
+				}
+				console.log("dictionary: " + JSON.stringify(dictionary));
 
-					allNearbyPokemon = allNearbyPokemon.filter( function( el ){ return (typeof el !== "undefined"); } );
-
-					// Assemble dictionary using our keys
-					var dictionary = {};
-
-					// take closest 9 (or fewer if not available; sentinel indicated by pokemonId == 0)
-					var j;
-					for (j = 0; j < 9; j++) {
-
-						if (j < allNearbyPokemon.length) {
-							dictionary["Pokemon" + (j + 1) + "Id"] = allNearbyPokemon[j].pokemonId;
-							dictionary["Pokemon" + (j + 1) + "ExpirationTime"] = allNearbyPokemon[j].pokemonExpirationTime;
-							dictionary["Pokemon" + (j + 1) + "Distance"] = allNearbyPokemon[j].pokemonDistance;
-							dictionary["Pokemon" + (j + 1) + "Bearing"] = allNearbyPokemon[j].pokemonBearing;
-						} else {
-							dictionary["Pokemon" + (j + 1) + "Id"] = 0;
-							dictionary["Pokemon" + (j + 1) + "ExpirationTime"] = 0;
-							dictionary["Pokemon" + (j + 1) + "Distance"] = 0;
-							dictionary["Pokemon" + (j + 1) + "Bearing"] = 0;
-							break;
-						}
+				// Send to Pebble
+				MessageQueue.sendAppMessage(dictionary,
+					function(e) {
+						console.log("AppMessage sent to Pebble successfully!");
+					},
+					function(e) {
+						console.log("Error sending AppMessage to Pebble!");
 					}
-					console.log("dictionary: " + JSON.stringify(dictionary));
-
-					// Send to Pebble
-					MessageQueue.sendAppMessage(dictionary,
-						function(e) {
-							console.log("AppMessage sent to Pebble successfully!");
-						},
-						function(e) {
-							console.log("Error sending AppMessage to Pebble!");
-						}
-					);
-				} else {
+				);
+
+				// TODO: much better error checking???
+				if (json.pokemon.length <= 0) {
```

Comment:
This is really the only change -- everything else is adjusting whitespace. I simply moved this conditional from line 74. The end result is that a dictionary of pokemon will be sent on *every* update, which _should_ clear the existing pokemon list if there are none. Once that's done, another message will be sent to display the string "No Pokemon are nearby" if there are in fact no pokemon in the area.